### PR TITLE
Task disconnected

### PR DESCRIPTION
**Pull request by pablofelix**
Steps:
1- Create a cluster with Hive
2- Add the variables in infrastructure_configuration.yaml
registryCA: <CA certificate>
registry_pull_secret: <registry_pull_secret>
3- Execute the task: 
./run_robot_test.sh --test-case tasks/Tasks/provision_self_managed_cluster.robot --extra-robot-args '--variablefile infrastructure_configuration.yaml' --include self_managed_disconnect --skip-oclogin
4- Follow RHODS official documentation to install/upgrade RHODS in a disconnected cluster

### CODE REVIEW

File: docs/infrastructure_configuration_variables.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+| `master_node_instance_type` | The AWS instance type to use for the master nodes. | `m5.xlarge` | `all` |
+| `master_node_replicas` | The number of master nodes to create. | `3` | `all` |
+| `aws_region` | The AWS region to use for the cluster. | `us-east-1` | `aws` |
+| `pull_secret` | The pull secret to use for the cluster. | `""` | `all` |
+| `ssh_key` | The SSH key to use for the cluster. | `""` | `all` |
+| `AWS_ACCESS_KEY_ID` | The AWS access key ID. | `""` | `aws` |
+| `AWS_SECRET_ACCESS_KEY` | The AWS secret access key. | `""` | `aws` |
+| `release_image` | The OpenShift image to use for the cluster. | `"quay.io/openshift-release-dev/ocp-release:4.10.42-x86_64"` | `all` |
+| `gcp_region` | The GCP region to use for the cluster. | `us-central1` | `gcp` |
+| `gcp_project_id` | The GCP project ID to use for the cluster. | `""` | `gcp` |
```
Comment (bdattoma) [inline docs/infrastructure_configuration_variables.md:23]: duplicated some lines after

File: docs/infrastructure_configuration_variables.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+| `hive_cluster_name` | The name of the cluster. | `rhods{provider}` | `all` |
+| `hive_claim_name` | The name of the claim. | `rhods{provider}claim` | `all` |
+| `image_set` | The image set name to use for the cluster. | `rhods-openshift` | `all` |
+| `aws_region` | The AWS region to use for the cluster. | `us-east-1` | `aws` |
+| `worker_node_instance_type` | The instance type to use for the worker nodes. | `m5.xlarge` | `all` |
+| `worker_node_replicas` | The number of worker nodes to create. | `2` | `all` |
+| `master_node_instance_type` | The AWS instance type to use for the master nodes. | `m5.xlarge` | `all` |
+| `master_node_replicas` | The number of master nodes to create. | `3` | `all` |
+| `aws_region` | The AWS region to use for the cluster. | `us-east-1` | `aws` |
+| `pull_secret` | The pull secret to use for the cluster. | `""` | `all` |
```
Comment (bdattoma) [inline docs/infrastructure_configuration_variables.md:17]: what does it mean? Pull secret for pulling what exatcly?

File: tasks/Resources/Provisioning/Disconnect/icsp.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: generic-0
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - registry.rh-ods.com:8443/openshift
```
Comment (bdattoma) [inline tasks/Resources/Provisioning/Disconnect/icsp.yaml:9]: could this become a parameter?

File: tasks/Resources/Provisioning/Hive/disconnect.robot [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ Disconnect Cluster
+    Oc Create    kind=ConfigMap    namespace=openshift-config
+    ...    src=tasks/Resources/Provisioning/Disconnect/disconnect.yaml
+    ...    template_data=${infrastructure_configurations}
+    Oc Patch     kind=Image    name=cluster    api_version=config.openshift.io/v1 
+    ...    src={"spec":{"additionalTrustedCA":{"name":"registry-config"}}}
+    Oc Delete   kind=Secret   name=pull-secret    namespace=openshift-config
+    Oc Create    kind=Secret    namespace=openshift-config
+    ...    src=tasks/Resources/Provisioning/Disconnect/registry_pull_secret.yaml
+    ...    template_data=${infrastructure_configurations}
```
Comment (bdattoma) [inline tasks/Resources/Provisioning/Hive/disconnect.robot:11]: same template for cluster creation and registry secret creation?

File: tasks/Tasks/provision_self_managed_cluster.robot [hunk 5]
```diff
@@ -16,7 +17,7 @@ Provision Self-Managed Cluster
     Provision Cluster
     Wait For Cluster To Be Ready
     Claim Cluster
-    Login To Cluster
+    # Login To Cluster
```
Comment (bdattoma) [inline tasks/Tasks/provision_self_managed_cluster.robot:20]: isn't login necessary anymore?

File: tasks/Tasks/provision_self_managed_cluster.robot [hunk 6]
```diff
@@ -36,3 +37,9 @@ Delete GPU Node From Self-Managed AWS Cluster
     [Tags]    gpu_node_aws_self_managed_deprovision
     Login To Cluster
     Delete GPU Node In Self Managed AWS Cluster
+
+Disconnect Self-Managed Cluster
+    [Documentation]    Disconnect a self-managed cluster
+    [Tags]    self_managed_disconnect
+    Login To Cluster
```
Comment (bdattoma) [inline tasks/Tasks/provision_self_managed_cluster.robot:44]: is this `Login To Cluster` refers to loggin into Hive right? If so, could you make the kw explicitly mentioning that?

Comment (pablofelix) [reply tasks/Tasks/provision_self_managed_cluster.robot]: Deleting Login To Cluster to allow the ability to choose between different login options and avoid conflicts

File: tasks/Tasks/provision_self_managed_cluster.robot (hunk 5 above)
Comment (pablofelix) [reply tasks/Tasks/provision_self_managed_cluster.robot]: This was deleted by mistake

File: tasks/Resources/Provisioning/Hive/disconnect.robot (hunk 4 above)
Comment (pablofelix) [reply tasks/Resources/Provisioning/Hive/disconnect.robot:11]: Different variables but all included in the same yaml file devoted to infrastructure variables. To use this yaml file was decided by the team some weeks ago

File: docs/infrastructure_configuration_variables.md (hunk 1 above)
Comment (pablofelix) [reply docs/infrastructure_configuration_variables.md]: This lines were added by mistake

File: docs/infrastructure_configuration_variables.md (hunk 2 above)
Comment (pablofelix) [reply docs/infrastructure_configuration_variables.md]: It is the cluster's default pull secret

File: tasks/Resources/Provisioning/Disconnect/icsp.yaml (hunk 3 above)
Comment (pablofelix) [reply tasks/Resources/Provisioning/Disconnect/icsp.yaml:9]: For now, we need to use always this repo

File: tasks/Tasks/provision_self_managed_cluster.robot (hunk 6 above)
Comment (bdattoma) [reply tasks/Tasks/provision_self_managed_cluster.robot]: so, I assume this refers to the Hive cluster right?

**Review by jgarciao — APPROVED** [review]
lgtm

**Review by mwaykole — APPROVED** [review]

**Review by tarukumar — APPROVED** [review]
LGTM

**Pull request merged**